core/src/main/java/com/linecorp/armeria/server/AnnotatedServiceBindingBuilder.java

```diff
@@ -0,0 +1,213 @@
+        return (AnnotatedServiceBindingBuilder) super.requestTimeout(requestTimeout);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder requestTimeoutMillis(long requestTimeoutMillis) {
+        return (AnnotatedServiceBindingBuilder) super.requestTimeoutMillis(requestTimeoutMillis);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder maxRequestLength(long maxRequestLength) {
+        return (AnnotatedServiceBindingBuilder) super.maxRequestLength(maxRequestLength);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder verboseResponses(boolean verboseResponses) {
+        return (AnnotatedServiceBindingBuilder) super.verboseResponses(verboseResponses);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder requestContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.requestContentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder responseContentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.responseContentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreview(int length) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreview(length);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreview(int length, Charset defaultCharset) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreview(length, defaultCharset);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder contentPreviewerFactory(ContentPreviewerFactory factory) {
+        return (AnnotatedServiceBindingBuilder) super.contentPreviewerFactory(factory);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder accessLogFormat(String accessLogFormat) {
+        return (AnnotatedServiceBindingBuilder) super.accessLogFormat(accessLogFormat);
+    }
+
+    @Override
+    public AnnotatedServiceBindingBuilder accessLogWriter(AccessLogWriter accessLogWriter,
+                                                          boolean shutdownOnStop) {
+        return (AnnotatedServiceBindingBuilder) super.accessLogWriter(accessLogWriter, shutdownOnStop);
+    }
+
+    /**
+     * Registers the {@link ServiceConfigBuilder} that is created from this Builder and returns
+     * {@link ServerBuilder}.
+     * @param service annotated service object to handle incoming requests matching path prefix, which
+     *                  can be configured through {@link AnnotatedServiceBindingBuilder#pathPrefix(String)}.
+     *                  If path prefix is not set then this service is registered to handle requests matching
+     *                  {@code /}
+     * @return {@link ServerBuilder} to continue building {@link Server}
+     */
+    public ServerBuilder buildAnnotated(Object service) {
```

Comment:
How about `build()` as we do in `ServiceBindingBuilder`?

Reply:
OK. I used this name because it would be clear to distinguish building annotated service and building server. I can make this change.

Reply:
For ex, it would look like:

```
  ServerBuilder sb = Server.builder();
  sb.annotatedService()                                    // Returns an instance of this class
    .requestTimeout(5000)
    .maxRequestLength(8192)
    .exceptionHandler((ctx, request, cause) -> HttpResponse.of(400))
    .pathPrefix("/foo")
    .verboseResponses(true)
    .contentPreview(500)
    .build(new Service())                         // Return to the ServerBuilder.
    .build();
```

Reply:
Yes, I have thought about it which `build(..)` is called twice consecutively.
Currently, we do this for `route()`:
```java
Server.builder()
      .route()
      .get("/")
      .build((ctx, req) -> HttpResponse.of(200))
      .build();
```
If we rename `buildAnnotated`, it will be:
```java
Server.builder()
      .annotatedService()
      .build(new MyObject())
      .build();
```
Which is a little weird. 😄 
But there's another chance that a user uses it with the variable:
```java
final AnnotatedServiceBindingBuilder annotatedBuilder = Server.builder().annotatedService();
annotatedBuilder.build(new MyObject());
```
I'm not sure which one is better honestly. 😄 But I still believe that we need to use same method name. WDYT?

Reply:
Thanks for the explanation. This totally makes sense. I will make this change along with adding `VirtualHostAnnotatedServiceBindingBuilder` and will notify you again for review.